src/primitive.ts

```diff
@@ -7,6 +7,12 @@ export const booleanP: TypeProxy<boolean> = (value) => {
     : { success: false, error: ParseError.simpleError(value, 'a boolean') };
 };
 
+export const dateP: TypeProxy<Date> = (data) => {
+  return (Object.prototype.toString.call(data) === '[object Date]' && data instanceof Date && !(isNaN(data.valueOf())))
+    ? { success: true, value: data }
+    : { error: ParseError.simpleError(data, 'a date'), success: false };
```

Comment:
Just for consistency with others:
```suggestion
    : { success: false, error: ParseError.simpleError(data, 'a date') };
```
Also for nitpicky consistency things, could use `= (value) => {` cause all the other ones do

Reply:
Done 👍